### ISSUE TITLE
autorelabel fails on Vagrant image for VirtualBox

### ISSUE THREAD

**Issue by schlomo** [CentOS/sig-cloud-instance-build#100]
Clone of https://bugs.centos.org/view.php?id=13213 which I put there because I didn't know better, thanks to @lpancescu for the intro to this repo.

We stumbled upon this in https://github.com/rear/rear-workshop-osbconf-2016/issues/4#issuecomment-286515264 and you can use the Vagrantfile in https://gist.github.com/schlomo/b532ba9bca87ea40d922d90e62b7338c as a test case.

@lpancescu The reason to use autorelabel in the VM is that we use this VM to do backup and restore tests so that we actually restore the VM's content into a new VM with the help of https://github.com/rear/rear and then we need to trigger autorelabel to restore the labels upon the first boot.

**Comment by lpancescu** [comment]
A serial console could be useful in some situations, and VirtualBox is the only one with a problem. Perhaps we could also enable the serial port for our VirtualBox image, and leave the cable disconnected. I hope to get some feedback from other VirtualBox users before removing the serial console from all virts.

https://lists.centos.org/pipermail/centos-devel/2017-May/015843.html

**Comment by schlomo** [comment]
Is there a way to find out if the serial port is actually there? Then you could decide dynamically. If there is no serial port then don't configure serial console.

Maybe `dmesg | grep ttyS0` is good enough?

**Comment by lpancescu** [comment]
@schlomo Could you please try to enable the serial port from your Vagrantfile? Just as a test to see if the mere presence of a disconnected port would fix the problem. Something like:

```ruby
config.vm.provider "virtualbox" do |v|
  v.customize ["modifyvm", :id, "--uart1", "0x3f8", "4", "--uartmode1", "disconnected"]
end
```
I hope I got the syntax right.

The console is a kernel boot parameter, set during the installation - not much we can do there.

**Comment by schlomo** [comment]
Yes, that also helped. However, I see this very problematic as the user will have to know about it. A simple `config.vm.box = "centos/7"` won't be enough any more.

How about you modify the grub config during the installation if the VM runs on VirtualBox?

In any case I hope that this is only a temporary workaround till upstream removes the requirement of the tty in the rhel-autorelabel unit.

**Comment by lpancescu** [comment]
@schlomo The installation is performed just once in `qemu-kvm`, and the resulting disk image is converted to several formats, for each Vagrant provider that we support. This was just a quick test to see if the presence of a disconnected serial port would prevent a hang during relabeling - it will be fixed on our side, without the users needing to make any changes to their Vagrantfile.

**Comment by schlomo** [comment]
@lpancescu thanks a lot. Maybe have some code modify the grub config at first boot under VirtualBox?

Something like this could already be a workaround:
```
if ! dmesg | grep -q ttyS0 ; then
      sed -i -e 's/ console=[^ ]\+\?//g' /etc/default/grub
      grub2-mkconfig -o /boot/grub2/grub.cfg
fi
```

**Comment by lpancescu** [comment]
@schlomo The Vagrant images v1704 were already released, just not visible on Atlas yet (I had built the release candidates this weekend). I'm afraid you'll have to wait until v1705, at the beginning of June.
